include/maidsafe/crux/socket.hpp

```diff
@@ -253,34 +254,32 @@ inline void socket::close() {
     keepalive_timer.stop();
     transmit_queue.shutdown();
 
+    auto m = multiplexer;
```

Comment:
Oh, I think I see what you mean, but it still won't be a problem. As far as I know, we're giving the same kind of thread safety guarantees in crux as [asio does](http://www.boost.org/doc/libs/1_57_0/doc/html/boost_asio/overview/core/threads.html). That is, one instance of a crux::socket can't be called from multiple threads concurrently (so no one can reset the socket::multiplexer variable between lines 252 and 257. That said, two or more instances can be called concurrently, that is why the multiplexer will need to be thread safe.